### ISSUE TITLE
Enable nginx recommended proxy settings

### ISSUE THREAD

**Issue by EricTheMagician** [ibizaman/selfhostblocks#114]
I was poking around and just wanted to make a comment.

I think you should enable the recommend proxy settings for nginx:
[services.nginx.recommendedProxySettings](https://search.nixos.org/options?channel=23.11&from=50&size=50&sort=relevance&type=packages&query=services.nginx)

There's a few other ones on that same note that are interesting to enable as well.
[services.nginx.recommendedZstdSettings](https://search.nixos.org/options?channel=23.11&from=50&size=50&sort=relevance&type=packages&query=services.nginx)
[services.nginx.recommendedTlsSettings](https://search.nixos.org/options?channel=23.11&from=50&size=50&sort=relevance&type=packages&query=services.nginx)
[services.nginx.recommendedOptimisation](https://search.nixos.org/options?channel=23.11&from=50&size=50&sort=relevance&type=packages&query=services.nginx)
[services.nginx.recommendedGzipSettings](https://search.nixos.org/options?channel=23.11&from=50&size=50&sort=relevance&type=packages&query=services.nginx)
[services.nginx.recommendedBrotliSettings](https://search.nixos.org/options?channel=23.11&from=50&size=50&sort=relevance&type=packages&query=services.nginx)

**Comment by ibizaman** [comment]
You're right, those should be added. Or there should be a comment why those are not used.
